packages/cli/src/commands/deploy/helper.ts

```diff
@@ -40,114 +40,105 @@ type GetPackageName = (params: {
   dir: string
 }) => Promise<{ error: false; name: string } | { error: true }>
 
-type CreateFieldTypeFunc = (params: {
-  name: string
-  body: unknown
-  client: StoryClientType
-  options: ManifestOption[] | undefined
-}) => Promise<FieldType>
-
-export const getPackageName: GetPackageName = async ({
-  name,
-  skipPrompts,
-  dir,
-}) => {
-  if (typeof name === 'string' && name !== '') {
-    return { error: false, name }
-  }
-
-  const packageJsonName = getPackageJsonName(dir)
-
-  if (skipPrompts === false) {
-    const nameFromPrompt = await promptName({
-      message: packageNameMessage,
-      initialValue: packageJsonName,
-    })
-
-    return {
-      error: false,
-      name: nameFromPrompt,
-    }
-  }
-
-  if (isValidPackageName(packageJsonName)) {
-    return { error: false, name: packageJsonName }
-  }
-
-  return { error: true }
-}
-
 // TODO: move all side effects to the deploy function
 export const upsertFieldPlugin: UpsertFieldPluginFunc = async (args) => {
   const { packageName, skipPrompts, token, output, dir, scope } = args
 
   const manifest = loadManifest()
 
-  const storyblokClient = StoryblokClient({ token, scope })
-
   printManifestOptions(manifest?.options)
 
+  const storyblokClient = StoryblokClient({ token, scope })
+
   console.log(bold(cyan('[info] Checking existing field plugins...')))
 
   const allFieldPlugins = await storyblokClient.fetchAllFieldTypes()
-  const fieldPlugin = allFieldPlugins.find(
+  const fieldPluginFound = allFieldPlugins.find(
```

Comment:
I kept looking for all the existing fields (from a specific scope) the user has access to because in case the creation fails due to name collision during creation time, and it belongs to the user performing the deployment, we can proceed with the update flow but then the `id` of the existing field plugin would be required.

Reply:
This seems reasonable, but I think this code will fetch field plugins from that specific `scope` 🤔
What if... the scope is currently `organization`, but the same field plugin exists in the `my-plugins` scope?

Anyway, this `fetchAllFieldTypes` fetches page 1 to the last page until it doesn't fetch anything. Initially I wrote it that way because there is no way to check if a field plugin exists or not. But I think it's a bit wasteful. Now that we're moving onto a new way ("create first, do something else if failed" approach), it might be just better to remove this part... What do you think? 🤔

Reply:
Oh, I just re-read the code and if we remove this part, then there is no way to update the existing field plugin 🤔 Then, do you think we should fix the way to fetch all the field plugins (from all the scopes)? I'm not too sure if I'm understanding this correctly.

Reply:
> Oh, I just re-read the code and if we remove this part, then there is no way to update the existing field plugin 🤔 

Yeah, that's it. 

Imagine a scenario where the creation failed due to a naming collision. 
Then, in such a case, we'd need to know if we should ask the user to rename it and try to deploy a new field plugin or if the user can just try to update it since it's owned by him/her.

To know which path to follow, we should then first try to fetch all the user's plugins, and then, if one of the fetched plugins is the same as the one being deployed, we can assume it belongs to the user, and then, now we have its ID, offers him to update it.

So, in case of creation failure, we'd need to discover if the plugin belongs to the user or not and if yes, to provide him an update flow, so, why not do it before, you know?! To avoid failing the creating time in the first attempt and have the required ID to follow with the update flow.

**The actual approach is:**
- Discover beforehand if the user owns the plugin (no matter in which scope it's deployed)
   - If yes then go to the update flow
      - Allows the user to rename and deploy a new plugin or just to proceed with the update since we have the ID for updating it.
   - if not, then try to create
      - if the creation works, then it's good
      - if the creation fails due duplicated name error, then ask the user if he/she wants to rename it and try again.
      
 In this approach, we avoid failing the first time in case the existing plugin is owned by the user.
 
 **The other approach would be:**
 
 - try to create a new field plugin
    - If it works, then it is good.
    - If it fails due duplicated name error, then we need to discover first if this plugin belongs to the user
       - If it belongs, then go to the update flow
          - Allows the user to rename and deploy a new plugin or just to proceed with the update since we have the ID for updating it.
       - If it doesn't belong, then ask the user if he/she wants to rename it and try again.
       
The same result as in the first approach but now it fails before suggesting the update flow and has one level more.

> Then, do you think we should fix the way to fetch all the field plugins (from all the scopes)? I'm not too sure if I'm understanding this correctly.

I considered this approach, but in the end, it wouldn't solve all the cases with the duplicated names issue since even fetching from all the scopes, would return only plugins owned by me or within the Partner Portal or Organization I belong to, so I decided on keeping this the way it's.

I mean, maybe fetching from all the scopes would reduce the chances of facing naming collision but it would not solve all the cases and I was also a little concerned about doing a lot of requests in a short period of time and ending by reaching the request/sec threshold, so I decided to keep the way it is.

What do you think about this all, @eunjae-lee?

Reply:
Thanks for the explanation. It totally makes sense. Let's keep it as-is!

It'd be nice to have sort of a flow chart somewhere around this deploy command since it's the most complex thing in this repository, in my opinion. (we already have it with tldraw but you could suggest another tool if you'd like to)

Reply:
I need to re-read this :D but to the last comment, maybe we can even have it inside the CLI readme? github markdown should support creating diagrams? that way it might be easier to maintain.

Reply:
just to clarify `allFieldPlugins` are all field plugins of a user from all scopes? if so could we maybe rename the variable to make it slightly easier to read?

Reply:
Hey @BibiSebi, I've just checked it in depth to give you a solid reply.

So, the fetched plugins will vary according to the chosen scope (My Plugins, Partner Portal, or Organization).

- If the user selects to deploy it to the `My Plugin` scope, then all the plugins the user has access to will be returned, meaning all the plugins created by him/her or plugins created by others (but in the same shared Partner Portal or Organization account he belongs to).

- If the user selects to deploy it to the `Partner Portal` scope, then only plugins created directly in the Partner Portal he belongs to, are going to appear to him. So, for example, if you and I belong to the same Partner Portal account and you and I create one plugin each, I would be able to see the one I created and also the one you have created. I wouldn't be able to see any of the plugins created directly in the `my plugin` or `organization` scopes. Only the ones created directly through the Partner Portal page.

- If the user selects to deploy it to the `Organization` scope, then only plugins created directly in the Organization he belongs to, are going to appear to him. So, for example, if you and I belong to the same Organization account and you and I create one plugin each, I would be able to see the one I created and also the one you have created. I wouldn't be able to see any of the plugins created directly in the `my plugin` or `Partner Portal` scopes. Only the ones created directly through the Organization page.

So, in the end, it's not `all field plugins` but the ones that are visible/accessible to me according to the scope I chose to perform the deployment.

But, to be sincere, I don't know a better name I could give to it. Maybe `scopeFieldPlugins`? 🤔 
What do you think, Bibi?

Reply:
scopeFieldPlugins sounds good

Reply:
The change was applied, @BibiSebi. 🙌